apis/v1beta1/validation/httproute.go

```diff
@@ -287,3 +289,39 @@ func hasExactlyOnePrefixMatch(matches []gatewayv1b1.HTTPRouteMatch) bool {
 
 	return true
 }
+
+// validateHTTPParentRefs validates that if ParentRefs includes 2 or more references
+// to the same parent (based on kind, name, and namespace), those ParentRefs must
+// specify unique SectionName values.
+func validateHTTPParentRefs(parentRefs []gatewayv1b1.ParentReference, path *field.Path) field.ErrorList {
+	var errs field.ErrorList
+	if len(parentRefs) <= 1 {
+		return nil
+	}
+	type sameKindParentRefs struct {
+		name      gatewayv1b1.ObjectName
+		namespace gatewayv1b1.Namespace
+		kind      gatewayv1b1.Kind
+	}
+	m := make(map[sameKindParentRefs][]gatewayv1b1.SectionName)
+	for i, p := range parentRefs {
+		if p.Namespace == nil {
+			p.Namespace = new(gatewayv1b1.Namespace)
+		}
+		if p.Kind == nil {
+			p.Kind = new(gatewayv1b1.Kind)
+		}
+		if p.SectionName == nil {
+			p.SectionName = new(gatewayv1b1.SectionName)
+		}
+		t := &sameKindParentRefs{name: p.Name, namespace: *p.Namespace, kind: *p.Kind}
```

Comment:
What's the point of getting a pointer to the struct if it is always dereferenced in the subsequent usages?

Reply:
My mistake.. fixed.